### ISSUE TITLE
Code of Conduct and enforcement. 

### ISSUE THREAD

**Issue by k8hertweck** [swcarpentry/board#111]
Moving discussion of the SWC Code of Conduct (CoC), communication in the community, and enforcement of expectations (following [this thread from Discuss list](http://lists.software-carpentry.org/pipermail/discuss_lists.software-carpentry.org/2016-May/004414.html)).

**Comment by embray** [comment]
CC @matthew-brett who started the original thread.

**Comment by codersquid** [comment]
background: I've helped with code of conduct violations in another group.

Moderating Behavior

In online behavior, once a thread becomes inflamed, a leader should step in, establish the community norms, and ask for that part of the thread to halt. An inflamed conversation should not silently die, because that does not reveal the community norms to lurkers. If something needs to be handled, the leader should let the community know. Example:

Everyone, please stop <behaviors>. These <behaviors> do not follow the code of conduct. We are taking this offline and handling it. <a real apology>.

Flipping the moderation bit on a topic or posters until things cool down is also good. That way someone won't make the mistake of posting in anger before seeing the request to stop.

I feel the criticism about the code of conduct being useless due to it not being engaged within 24 hours is a little harsh, if you don't have moderators to help cover a 24 hr period. Maybe when you work things out you can put people on moderator duty for the mailing list.

Accepting Reports of Behavior

* More than one person should handle reports.
* For however you accept reports, you need to establish a precedence of privacy and confidentiality. For example, maybe have a mailing list that accepts reports, and it can accept posts from everyone, but only the conduct committee has access to it. (This does not provide future-privacy. I haven't worked out a good protocol for that.)

Provide scripts for awkward interactions.

**Comment by wrightaprilm** [comment]
It seems like there are three issues that have come up, here:

1. If the particular situation warranted a person being thrown off the mailing list
2. If the steps needed to remove someone from a SWC/DC space are formal enough
3. If a CoC helps us get the behavior we want to see in participants

Firstly, I think a CoC that outlines what is not considered professional and appropriate behavior in SWC/DC communications and events is needed. We don't know the backgrounds of all participants and instructors, and so what is not permitted needs to be specified. What is considered acceptable behavior in a grad student office is different than in an undergrad classroom than in a shareholders meeting, and all these contexts are ones in which our volunteers and learners may be operating. So the clear delineation of 'No, you can't do that', I think is a good one. Whether there needs to be a separate document of positive behaviors, I have no strong feelings. 

In terms of formalizing procedures to remove someone, I think that's hard. Generally harassment on a listserve doesn't rise to the level of any sort of criminal harassment. But that doesn't mean someone shouldn't step in and put someone who is being a jerk on a time-out. Perhaps we need to have a handling board, as @codersquid suggests so that the options are more varied than just kicking someone off the list. There is still going to be an issue at workshops, though, where someone is being disruptive and there are only two instructors, and this will always limit our ability to act.

For this specific individual ... I can't say what's appropriate. She continued harassing me off-list, and therefore, I think Jonah has been very reasonable, but I think my rationality in this is compromised.

**Comment by embray** [comment]
> Firstly, I think a CoC that outlines what is not considered professional and appropriate behavior in SWC/DC communications and events is needed

I'd need to dig up my notes before I can be more specific, but when we discussed the code of conduct for the open Astronomy community (see http://python-in-astronomy.github.io/2016/code-of-conduct.html) an earlier draft included many specific examples of inappropriate behavior and phrasing.  After a long discussion we decided to take those out, for a number of reasons.  For example, by providing a limited set of examples it was felt that it weakened the CoC because it was too easy to focus on those examples and not the broader theme.

> Whether there needs to be a separate document of positive behaviors, I have no strong feelings.

A separate document providing positive *and* negative examples is not necessarily a bad idea.

More broadly, some context for why a CoC is important and useful (even if it's perceived to not have any enforcement teeth) could be useful.  Another development community I participate in right now unfortunately lacks a solid CoC because last time the issue was discussed, in my opinion not enough context was provided for why such a thing is good to have. If you're curious about how that went down I can discuss it more offline one-on-one, but suffice it to say the purpose of a Code of Conduct is not obvious to everyone, unfortunately...

**Comment by bradenowsley** [comment]
I would simply like to state that all of this drama of the last day or so almost moved me to get off the Discuss list. I would strongly encourage SWC to enforce a principle of utility on the Discuss list and move the Facebook style drama somewhere else. I don't know about everyone else but I already don't have time for everything that ends up there.

**Comment by matthew-brett** [comment]
It would be great if it were possible to have a healthy community without any conflict, but I personally have never seen one.  The problem is that if the community tries to avoid conflict the level of discussion drops very quickly, and becomes boring and irrelevant.   For characteristic examples of the health in robust discussion, see the quotes in this blog post : http://nipyworld.blogspot.com/2012/02/what-is-meritocracy-in-open-source.html

**Comment by jduckles** [comment]
This has been moderated and moved. I am getting back-channel emails flooding in that there is too much Discuss traffic. Thank for your understanding.

@matthew-brett said:
> Yes, but moving the discussion to a github issue also has costs, which
> are of course soft costs:
> 
> 1) Moving to an issue reduces the number of people who join the
> discussion, so the result is less likely to reflect the whole
> community;
> 2) It carries the potential for implying that the discussion is
> somehow specialized and not general.
> 
> For example on the spawned thread there is already a comment that
> implies that the poster would want this kind of discussion removed
> from the list entirely, on the basis that it's not useful ("I would
> strongly encourage SWC to enforce a principle of utility on the
> Discuss list").    That's obviously a big risk here, in the sense that
> no or little discussion could mean a strong bias for the status quo,
> and I think that would be major loss of an opportunity to make things
> better.
> 
> So, yes, sure I can see that there will be times when the discussion
> becomes specialized and only requires a few people to join in - the
> same kind of time where you spin off a sub-committee to deal with a
> particular topic.    I'm not even disagreeing that the current
> discussion could become that.  I was only saying that it might need
> some justification or explanation, in view of the costs.
> 
> Cheers,
> 
> Matthew

**Comment by matthew-brett** [comment]
On Fri, May 6, 2016 at 12:44 PM, Jonah Duckles <notifications@github.com> wrote:
>
> This has been moderated and moved. I am getting back-channel emails flooding in that there is too much Discuss traffic. Thank for your understanding.

I assume that another email questioning whether that's a sensible
thing to do will also be moderated and moved?

Best,

Matthew

**Comment by katyhuff** [comment]
If I recall the history correctly, the whole discuss listhost was
originally created to be the place to corral the kinds of discussions that
cause the kind of email traffic that annoys people. The logic was that
there was a lot of traffic on the instructors list or whatever and
therefore there should be a listhost where people can discuss relevant but
non-essential topics. People uninterested in traffic would then be free to
get off the discuss list. Is that not the correct history @gvwilson ?

Anyway, that's just a history lesson though. The reality is that lots of
important things are on the discuss list nowadays, so people want it to
only contain important things and are now unwilling to unsubscribe, even if
they don't like email traffic.

Personally, I think this just means we should all use the slack channel
more...


On Fri, May 6, 2016 at 9:49 AM, Matthew Brett <notifications@github.com>
wrote:

> On Fri, May 6, 2016 at 12:44 PM, Jonah Duckles <notifications@github.com>
> wrote:
> >
> > This has been moderated and moved. I am getting back-channel emails
> flooding in that there is too much Discuss traffic. Thank for your
> understanding.
>
> I assume that another email questioning whether that's a sensible
> thing to do will also be moderated and moved?
>
> Best,
>
>
> Matthew
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/swcarpentry/board/issues/111#issuecomment-217495918>
>



-- 
http://katyhuff.github.com

**Comment by karawoo** [comment]
To propose a counterpoint, I personally am much more likely to weigh in on an active thread once it moves to GitHub. I am self conscious and inhibited about adding traffic to email threads and cluttering people's inboxes, but once a thread moves to GitHub (or Slack) I'm more comfortable adding my voice, knowing that the people it will reach are ones who have opted in to hearing it.

**Comment by matthew-brett** [comment]
On Fri, May 6, 2016 at 12:53 PM, Katy Huff <notifications@github.com> wrote:
>
> If I recall the history correctly, the whole discuss listhost was
> originally created to be the place to corral the kinds of discussions that
> cause the kind of email traffic that annoys people. The logic was that
> there was a lot of traffic on the instructors list or whatever and
> therefore there should be a listhost where people can discuss relevant but
> non-essential topics. People uninterested in traffic would then be free to
> get off the discuss list. Is that not the correct history @gvwilson ?
>
> Anyway, that's just a history lesson though. The reality is that lots of
> important things are on the discuss list nowadays, so people want it to
> only contain important things and are now unwilling to unsubscribe, even if
> they don't like email traffic.
>
> Personally, I think this just means we should all use the slack channel
> more...

Sure, but in this case, and especially given the previous discussion,
it seems like a pretty bad idea to moderate or remove my email.    I
hate to keep saying this, but being heavy-handed as the list moderator
has a cost.

I think you've got to decide how you're going to react to people
complaining about too much discussion on the discussion list.   If you
really think the right way to react is to reduce the discussion, you
should say that publicly and explain that.   I think that's a terrible
idea, because you'll get a boring list where people don't want to
speak up, but it's your call.

I don't think it's OK to just pull my email off the list without prior
warning.  I don't think there's anything repetitive about my email,
and the point seems to me substantial and worth discussing.

Best,

Matthew

**Comment by katyhuff** [comment]
On Fri, May 6, 2016 at 10:02 AM, Matthew Brett <notifications@github.com>
wrote:

>  I think that's a terrible
> idea, because you'll get a boring list where people don't want to
> speak up, but it's your call.
>
>
Actually, for the record, it's not my call. I'm not on the steering
committee anymore and had nothing to do with removing your email. I think
you have a point and I actually was adding some historical perspective that
(I thought) supported your point in some ways.



> I don't think it's OK to just pull my email off the list without prior
> warning. I don't think there's anything repetitive about my email,
> and the point seems to me substantial and worth discussing.
>

I don't disagree.

**Comment by matthew-brett** [comment]
On Fri, May 6, 2016 at 1:05 PM, Katy Huff <notifications@github.com> wrote:
>
> On Fri, May 6, 2016 at 10:02 AM, Matthew Brett <notifications@github.com>
> wrote:
>
> > I think that's a terrible
> > idea, because you'll get a boring list where people don't want to
> > speak up, but it's your call.
> >
> >
> Actually, for the record, it's not my call. I'm not on the steering
> committee anymore and had nothing to do with removing your email. I think
> you have a point and I actually was adding some historical perspective that
> (I thought) supported your point in some ways.

I'm sorry - I lazily replied to your email as I was mainly replying to
what you were saying.   I didn't at all mean to blame you for removing
my email - and thanks for the perspective.

See you,

Matthew

**Comment by matthew-brett** [comment]
On Fri, May 6, 2016 at 1:10 PM, Matthew Brett <matthew.brett@gmail.com> wrote:
> On Fri, May 6, 2016 at 1:05 PM, Katy Huff <notifications@github.com> wrote:
>>
>> On Fri, May 6, 2016 at 10:02 AM, Matthew Brett <notifications@github.com>
>> wrote:
>>
>> > I think that's a terrible
>> > idea, because you'll get a boring list where people don't want to
>> > speak up, but it's your call.
>> >
>> >
>> Actually, for the record, it's not my call. I'm not on the steering
>> committee anymore and had nothing to do with removing your email. I think
>> you have a point and I actually was adding some historical perspective that
>> (I thought) supported your point in some ways.
>
> I'm sorry - I lazily replied to your email as I was mainly replying to
> what you were saying.   I didn't at all mean to blame you for removing
> my email - and thanks for the perspective.

Sorry again - I mistyped - I meant to say:

I lazily replied to your email as _if_ I was mainly replying to ...

Matthew

**Comment by wrightaprilm** [comment]
@embray wrote:

> I'd need to dig up my notes before I can be more specific, but when we discussed the code of conduct for the open Astronomy community (see http://python-in-astronomy.github.io/2016/code-of-conduct.html) an earlier draft included many specific examples of inappropriate behavior and phrasing. After a long discussion we decided to take those out, for a number of reasons. 

And to be clear, I wasn't saying that specific examples need to be added. I was just reacting to the statement that appeared on the listserve that the examples should all be positive. I disagree with that, because not everyone will have the context to realize that exclusionary jokes, for example, are not considered professional conduct for SWC.

**Comment by jiffyclub** [comment]
We try to be sensitive of people that don't have advanced email workflows. Long email threads may not be disruptive to some, but for a lot of people they are a significant increase in their daily email load and bury emails they actually need to see/respond to. Moving to GitHub is a simple way to allow discussion to continue while respecting people's inboxes. There's no evidence that the practice has any dampening affect on discussion.

I've never seen a discussion so important that it was necessary to allow it to be uncivil, or any discussion that was improved by incivility. I'm quite thankful for our CoC and the standard it sets for our community.

**Comment by matthew-brett** [comment]
On Fri, May 6, 2016 at 2:28 PM, Matt Davis <notifications@github.com> wrote:
>
> We try to be sensitive of people that don't have advanced email workflows. Long email threads may not be disruptive to some, but for a lot of people they are a significant increase in their daily email load and bury emails they actually need to see/respond to. Moving to GitHub is a simple way to allow discussion to continue while respecting people's inboxes. There's no evidence that the practice has any dampening affect on discussion.
>

Sorry - I probably wasn't being clear - I wasn't trying to discuss
moving to github issues here - I was trying to discuss whether it was
reasonable to pull my email without warning on the basis that there
was too much traffic.

> I've never seen a discussion so important that it was necessary to allow it to be uncivil, or any discussion that was improved by incivility. I'm quite thankful for our CoC and the standard it sets for our community.
>

On civil and otherwise discussion, I pointed this out on the thread
that spun off:

http://nipyworld.blogspot.com/2012/02/what-is-meritocracy-in-open-source.html

Cheers,

Matthew

**Comment by drj11** [comment]
I'm totally happy with any discussion being moved to github. Demanding a discussion stay on list means that 700 people have it land in their email inboxes.

**Comment by jiffyclub** [comment]
I don't think there is any shortage or discouragement of disagreement in the Software Carpentry universe. We have plenty of productive disagreement that has helped surface valuable insights for our community. But there is no reason to allow uncivil discussion in this space. The bottom line is this: if you can't be civil in an argument you are failing at communication and need to be silent until you can commit to doing better.

**Comment by matthew-brett** [comment]
On Fri, May 6, 2016 at 2:58 PM, Matt Davis <notifications@github.com> wrote:
>
> I don't think there is any shortage or discouragement of disagreement in the Software Carpentry universe. We have plenty of productive disagreement that has helped surface valuable insights for our community. But there is no reason to allow uncivil discussion in this space. The bottom line is this: if you can't be civil in an argument you are failing at communication and need to be silent until you can commit to doing better.
>

Sorry - I shouldn't have added that last part to my email, it was just
distracting.   I guess the discussion of whether there is or is not
enough space for disagreement should go elsewhere.

This just to return to my original question - is it acceptable to pull
(in this case my) email from the list on the basis of too much
traffic, and without warning?

Cheers,

Matthew

**Comment by karawoo** [comment]
When the thread has moved to GitHub, particularly after multiple people have requested it be moved, I think it is both acceptable and desirable that list moderators enforce this transition. So in my opinion, the answer is yes.

**Comment by wking** [comment]
On Fri, May 06, 2016 at 12:03:43PM -0700, Matthew Brett wrote:
> … is it acceptable to pull (in this case my) email from the list on
> the basis of too much traffic, and without warning?

I'm not familiar enough with Mailman to know what knobs are available.
@codersquid suggests thread and/or poster moderation [1], and those
both sound fine to me.  A moderator could post a capping message to
the thread (“moved to $NEW_CHANNEL, further posters will be
moderated”), and everybody gets on with business as usual.  Removing
an email that has already landed in the archives (is that what
happened here?) seem extreme.

[1]: https://github.com/swcarpentry/board/issues/111#issuecomment-217483972

**Comment by matthew-brett** [comment]
@karawoo - just to check - did you read my email?   My email was asking and discussing the general policy for moving threads to github - so not related to the open thread on github.

**Comment by karawoo** [comment]
I was responding to the question in your most recent comment: 

> is it acceptable to pull (in this case my) email from the list on the basis of too much traffic, and without warning?

Again, my opinion is yes.

**Comment by matthew-brett** [comment]
Sorry - I think I'm not understanding.   You have read my email, which is about the general policy of moving threads to github.  You agree that it is not related to this thread, which is about "code of conduct and enforcement".  Nevertheless you think it was OK to pull the email.   I guess then your previous explanation does not apply - so can you explain more?

**Comment by wking** [comment]
I've spun off “where should high-volume discussion go?” into #112, so
we can separate it from “what happens to high-volume threads in a
low-volume channel?” issue (which seems to be the focus here, maybe
update the title?).

**Comment by matthew-brett** [comment]
No, I think this thread is meant to be about an entirely separate issue, which is code of conduct and enforcement - probably my fault - I must have got confused with my github thread email addresses.  So I suggest we drop the discussion of email filtering here.

**Comment by karawoo** [comment]
@matthew-brett This conversation has become highly derailed and in my view is no longer a good faith discussion, so no I will not explain more.

I'm happy to continue discussing issues of codes of conduct and enforcement.

**Comment by matthew-brett** [comment]
I completely agree it would be good to leave this particular thread as CoC and enforcement.

**Comment by SpacedGirl** [comment]
Not sure if I should be commenting given I was banned from a related thing [the Discuss list] 

My thoughts on the CoC issue and the discussion prior that resulted in it's mention is that there are two very disparate groups of people here [x2]. 
1) Those who like the CoC and do not want any talk to exist that they consider 'unprofessional' and non-civil and off-point [i.e. 'inflamed' threads] and others who have issues with it
2) Also those who want a heavy handed moderating approach to stop/decide issues and those that want a discussion to run its course.

What I'm left with based on this is:
-is there room for creating a non-behaviour moderated forum and open process one where undesirable talkers and topics [i.e. non 'professional' writing/thoughts] can be safe to exist?
[essentially an adults only sort of forum]

-S.G.

**Comment by jiffyclub** [comment]
The rest of the internet is open to you. Anything specifically within the SWC community needs to abide by our CoC.

**Comment by katyhuff** [comment]
@SpacedGirl 

I can't think of any productive work* that can't be accomplished in a safe** forum but can be accomplished in an unsafe one.

\* 'work,' here, refers to anything relevant to the software carpentry mission.
\*\* I have used the word 'safe' to refer to any forum regulated by our code of conduct. (civil discourse)

**Comment by SpacedGirl** [comment]
@jiffyclub 

I'm going to need further clarification as to what you mean in terms of my posting here. I'm confused by your answer.

**Comment by shwina** [comment]
@SpacedGirl, I don't think @jiffyclub's comment was in regards to your posting here. I think it addressed your suggestion for a forum that is

>  non-behaviour moderated

and

> open process one where undesirable talkers and topics can be safe to exist

by which, I assume, is meant a forum where the CoC is not enforced in its entirety?

**Comment by SpacedGirl** [comment]
@shwina 

Yes. A CoC exempted place for SWC related discussions.

**Comment by jiffyclub** [comment]
Software Carpentry will not create any space in which the Software Carpentry Code of Conduct does not apply. Full stop.

**Comment by katyhuff** [comment]
Yep.
This is well documented.

- http://software-carpentry.org/conduct/
- https://github.com/swcarpentry/board/blob/a522db4520fa18a9d8baefcaef6798220af59f29/strategy/strategic_plan.md#community

**Comment by SpacedGirl** [comment]
ok. I was presenting the option.
@katyhuff 
Thanks for providing the strategic plan. Looked through; think it'll help in my understanding of the structuring of everything here. Don't think I'd come across it before.

**Comment by ctb** [comment]
A few thoughts, all IMO -

I support @jduckles  in all of the actions I know about so far - placing the list in moderation, and banning S.G, and moving the thread to github.

I am happy with the SCF director being the initial arbiter of bad conduct and list policies, and most specifically in making the first call on whether someone is banned for CoC violations. There can be an appeals board that can be formed from Advisory Board or Steering Committee people and/or associated community members. The steering committee is elected by the community and can be assumed to represent them; the SCF director is hired by the committee to be the executive, and this is an executive or operational function that often needs to be dealt with quickly.

It would be good to articulate clearly and precisely the long-standing (but perhaps rarely specified) goal that this list is meant to be a low volume discuss list. I know @gvwilson (the creator of the list and founder of SCF) has struggled to maintain a balance between participation and deluge, and I know many, many people who have dropped off the list as the volume has grown.  To try to articulate it myself, the discuss list is one of the primary information channels for what is going on with Software Carpentry and its associated community of instructors (which now includes Data Carpentry and others), and should be kept for information updates and requests for links, not for broad and opinionated discussion.

Maybe we should have some advice to lurk on the list a bit to see how things are done in this community and on this list? We punt threads to github a few times a month IIRC. I don't know if S.G. would have benefited from a few weeks reading the list before jumping into the fray (I recall her saying that she'd joined just a few days ago somewhere).

There could be an address or a FAQ to re list policies, and then a resolution address for people who are wondering what they are and why. The list is simply not the right place to discuss list policies IMO; see next point.

And... sorry, this seems to now be a crossover comment to #112...

**Comment by SpacedGirl** [comment]
Re: @ctb and **"I don't know if S.G. would have benefited from a few weeks reading the list before jumping into the fray"**

I wouldn't have benefited. While I just joined a couple of weeks ago, I've tried reading into SWC stuff and often been confused as to functioning/structuring  and expected audience for things. Think just asking people to lurk will not have added benefit. It takes certain conversations to occur to inform people of the functioning of the group. One could watch for awhile and not read into the interactions what you'd like them to take from it or get much on the functioning/structuring of things due to the particular discussions being had .

**Comment by swaldman3** [comment]
My 2p, for what it's worth:

* I fully support having a CoC, and have no problem with the principles of the one that we have - but it is clearly intended to apply at workshops right now, and should probably be amended to include online behaviour. If nobody else gets to it, I will attempt this later in the weekend.

* I don't think there's much to be gained in simply restating the CoC in different language in another document. However, I wonder whether it would be worth having a "Welcome to SWC" document that introduces the community, incorporating this and also explaining things like the roles of the mailing list and github, and generally how to participate. This might also help with the confusion that SpacedGirl mentions in the comment above this. If such a document is written, care must be taken to ensure that it clearly signposts the CoC so that it is not seen as replacing it in anybody's mind.

* I fully support allowing people to contribute to SWC anonymously or pseudonymously.

* I welcome SpaceGirl's input, but I do think that the tone of it diverged from "appropriate for a professional audience" as specified in the CoC. It wouldn't be acceptable to walk up to a stranger at a scientific conference and swear at them or call them a fascist, so it shouldn't be acceptable on the SWC mailing list. As such, I think that unsubscribing SG was in line with the CoC, but that doing it without any warning was heavy-handed. I think that moderators must be allowed discretion in such matters, and am against defining moderation procedures too rigidly, but I would urge them to view bans as a last, rather than a first, resort. If SG still wants to contribute to the community, and is prepared to do it without being personally insulting, I would support welcoming them back onto the list.

**Comment by wrightaprilm** [comment]
I largely agree with everything you've said @swaldman3, and I'm glad you've made the change you did to the CoC. 

One thing that I'm unclear on is the relationship between the mailing list and the GitHub issue tracker. There's a person participating in this thread who was banned from the mailing list. I had to block this person from my personal email after receiving more messages from them that were ... let's just call them 'in violation of the code of conduct'. They seem to be behaving themself now, but if they were less inclined to do so, they would now have a way to access me that they didn't before. I could be wrong, but I don't think GitHub users can block other users from tagging them? In any event, it seems inconsistent to block someone from one space and not another. @jduckles, can you shed some light on this point?

**Comment by wking** [comment]
On Sat, May 07, 2016 at 06:49:11AM -0700, April Wright wrote:
> I could be wrong, but I don't think GitHub users can block other
> users from tagging them?

You can block and/or report-to-GitHub any user from their use page
(e.g. [1]) [2].  The icon is currently an octogon with an exclaimation
point inside it.  I'd guess blocks are per user-pair, and GitHub
investigates reports and takes stronger action as it sees fit, but I'm
not sure.

And there's always procmail or other mail filtering for folks who
consume GitHub via email, which works regardless of what GitHub
decides to support ;).

[1]: https://github.com/wking
[2]: https://github.com/blog/862-block-the-bullies

**Comment by SpacedGirl** [comment]
@wrightaprilm 

Now that's not an entirely accurate statement given that you emailed me last.

@wking 
Thanks for info. Find navigating github itself for info on stuff to be incredibly daunting/difficult. Should use the caveat that I haven't tried to do so recently. Changes might have occurred.

@swaldman3 
Think a welcome to SWC document is a perfect idea

One technical thought here due to @swaldman3's post. If one were a proceduralist, one would realize that I shouldn't have been banned on CoC grounds. Interesting conundrum.

**Comment by wrightaprilm** [comment]
Thanks for the info, @wking. Will be making use of it. 

There's still need for clarification about what spaces a ban covers, but exercising provided block utilities is an acceptable bandaid.

**Comment by jiffyclub** [comment]
We can also, if necessary, block a user entirely from the swcarpentry organization: https://help.github.com/articles/blocking-a-user-from-your-organization/ @wrightaprilm please let us (including @jduckles) know if you'd like us to take further action here.

**Comment by liz-is** [comment]
There were two objections to the Code of Conduct that were brought up on the discuss list that I don't see mentioned here:

- The legal / formal wording of the CoC may be offputting [1]
- Concepts of politeness/respect are influenced by dominant social ideas [2] -- may be linked to the idea of tone policing [3]

For the record, I am strongly in favour of a Code of Conduct and it makes me more likely to participate in SWC events and discussions. I think the benefit of the legal / formal wording of the CoC outweighs any downsides. I'm not sure how it would be possible to address the second point, but again I'm inclined to say that the benefits of asking for professional behaviour (according to the dominant social ideal of 'professional') are greater than any problems, and that it's possible to have professional behaviour without tone policing.

[1] http://lists.software-carpentry.org/pipermail/discuss_lists.software-carpentry.org/2016-May/004426.html
[2] http://lists.software-carpentry.org/pipermail/discuss_lists.software-carpentry.org/2016-May/004388.html
[3] http://geekfeminism.wikia.com/wiki/Tone_argument

**Comment by wking** [comment]
On Sun, May 08, 2016 at 07:56:26AM -0700, Liz Ing-Simmons wrote:
> - The legal / formal wording of the CoC may be offputting [1]

I am also personally fine with legal / formal wording.  But if someone
wants to write an informal summary (like the CC summaries? [1]), that
would be fine with me.

> - Concepts of politeness/respect are influenced by dominant social
>   ideas [2] -- may be linked to the idea of tone policing [3]

I agree that tone policing can be used to distract from discussion the
policer doesn't like.  But inflammatory language can also be used to
distract from discussion that the inflamer doesn't like, or just to
troll.  I'm fine letting the Executive Director, lead instructor, or
other channel-owner ban folks from channels they own as they see fit
(although I'd prefer objectionable material was left in the archives
with a clear flag stating why the material was objectionable and that
the poster was banned).  Folks who feel like banning is being abused
can appeal to the steering committee.  If they've alienated the
steering committee, they can try to find an intermediate that the
steering committee will listen to.  If they've alienated everyone in
SWC, well, the world is a big place [2].

[1]: http://creativecommons.org/licenses/by-sa/4.0/
[2]: https://github.com/swcarpentry/board/issues/111#issuecomment-217550365

**Comment by SpacedGirl** [comment]
Re: last two posts

Think that formal/legal/restrictive wording of a CoC is intimately tied into tone policing and the 'professional' concept. Essentially think that CoC's that are written in a restrictive way, lead to frequent referencing by individuals who think someone's out of bounds [often times its a matter of their conceptualizations of what the allowable social norms of a place are or what a respectful individual talks like]. The end effect is frequent silencing of conversations.

Think talk of inflammatory language is completely tied up in the tone policing concept. People who get accused of inflammatory/intimidating conversation are almost always on the lower end of the power spectrum [i.e. just a member and/or a student] and it's their ownership of their perspective [rather than referring to their authoritative position  --->which they almost never have] that gets judged/criticized and found in need of policing. Think it should be pointed out that using the word 'inflammatory' is inherently a biased word. It assumes that a person is trying to start arguments and problems with others. Use of the word 'intimidating' is also similarly problematic.

On the side of those who post loudly so to speak; it's been my experience that rarely do those sorts of individuals try to completely silence others, they just refuse to be silenced themselves.

In terms of my problems with what has occurred both on the list and here has less to do with social issues [had to be brought up due to the way people don't like my preferred way of discussing things......examples include: conversational vs professional talk; saying kick me off if you don't like it vs. saying I'm being bullied] but rather technical issues with the running of things here.

For instance, just referring to a very abstract CoC does not form an argument in and of itself. Any references to that document [by anyone; admin/committee/just a member] with respect to a poster's behaviour should also include which exact part of a post was problematic and how it violated the CoC. 

Fundamentally though my problem with this group as seen through my experience interacting with it is that judgments occur before explanations. That approach is antithetical to both logic and teaching itself.

**Comment by jiffyclub** [comment]
Software Carpentry expects professional behavior in all Software Carpentry spaces online and off because we are a professional organization. We are a non-profit and mostly run on volunteer work, but on a basic level we provide teaching services in exchange for money. What organization is going to invite us to give a workshop if they don't expect the instructors to be professional? What instructor is going to volunteer to teach if they don't expect their co-instructor to be professional? Trusting each other is a real and significant part of what makes Software Carpentry work.

Imagine that everything said here is going to be read by someone you will meet in person and to whom you will be teaching, or by someone you will have to teach alongside. Imagine that, because that is what is actually happening. If you can't engage with us on those terms, this is not the community for you.

**Comment by embray** [comment]
It's true that "tone policing" is used in vertical power structures to silence people on the lower end while ignoring the merits of their particular arguments or issues. So in that sense SpacedGirl isn't wrong.  And up to a point I'd have given SpacedGirl a pass for getting angry--[their first post on `[Discuss]`](http://lists.software-carpentry.org/pipermail/discuss_lists.software-carpentry.org/2016-May/004344.html) (that I saw) was a perfectly on-topic and constructive post. And the only follow-up to it was a completely off-topic (and incorrect) demand to not post pseudonymously--that would have made me angry too, different people have different levels of control when it comes to responding out of anger, unfortunately. I think after Jonah stepped in that should have been the end of it--but it's the sustained angry posting that is a problem.  Claiming a tone argument is not an excuse for sustained unprofessional behavior.

I would also add that there has been an effort on the alt-right lately to misuse the concept of "tone policing" as part of a bad-faith argument against what they call "political correctness" and community behavior standards (i.e. Codes of Conduct), so I'm a little wary of this argument.

> For instance, just referring to a very abstract CoC does not form an argument in and of itself. Any references to that document [by anyone; admin/committee/just a member] with respect to a poster's behaviour should also include which exact part of a post was problematic and how it violated the CoC.

I think this misunderstands the purpose and use of a Code of Conduct, and it's a misunderstanding I've seen before.  Part of the misunderstanding is in the phrasing--some people read "code" as a specific list of rules and laws, as in the Code of Hammurabi (or any legal code).  Some Codes of Conduct may be written in such a way, but many aren't.  Instead, the Code of Conduct sets basic expectations of tone and behavior, and exists to reassure anyone who might have any concerns about joining a community that the community has such standards.  It's not there so that someone can make a legal argument along the lines of "This posts violates title 3, section C of the Code of Conduct."   Referencing the CoC is not done to "form an argument".  It's done to remind someone what the standards of behavior are.  I think [Ubuntu's Code of Conduct](http://www.ubuntu.com/about/about-ubuntu/conduct) makes this point succinctly:

> This Code is not exhaustive or complete. It is not a rulebook; it serves to distil our common understanding of a collaborative, shared environment and goals. We expect it to be followed in spirit as much as in the letter.

**Comment by SpacedGirl** [comment]
RE: @embray 's post
Part 1 of your post:
I think you misunderstand the concept of tone policing. Refer to @liz-is 's post and the third reference in it for clarification. When doing so, please compare your statements [to follow in quotes] there to the sorts of ones mentioned in the Ref.
**"And up to a point I'd have given SpacedGirl a pass for getting angry"**
**"different people have different levels of control when it comes to responding out of anger, unfortunately. I think after Jonah stepped in that should have been the end of it--but it's the sustained angry posting that is a problem."**

Part 2 of your post:
You misunderstood what I meant. 
You are absolutely correct that a code of conduct isn't exhaustive/complete and in fact really can't be a rulebook unless it's a dictionary sized one.
My point was that when a person thinks a code of conduct violation has occurred in an online interaction; they should not just say so or refer a person to a very general statement but point out as well how the online words/statements violated the Code. Think there are now many examples of people doing so but only after I was banned. My point is that if a person wants to complain about a person's actions at all, they should also make sure to show what they are complaining about and how it is in violation of the CoC.
Hopefully that helps in clarifying my meaning. Not sure if that's the case. Thought I did ok with the explanation but c'est la vie when dealing with individuals other than oneself.

**Comment by SpacedGirl** [comment]
Re: My May 8, 2016 5: 57 PM EDT github comment [currently seen above]

Think my statements contained therein are of the most profound and successful I've ever had with respect to online facilitation and moderation of people and code of conducts [also relates to my philosophy on administrating over people/students/employees in general].

It's my take vs. 99.99% of the populations view on acceptable/correct/effective moderation. While I've been actively arguing with others for the last couple of years about such things, have yet to be able to sum up my views in a way I like when arguing these issues.

Anyways, the issues/examples presented by me are not simple in the least and even if one were to agree in principle they are very difficult to point out/challenge in action. I'm totally saving that for later use and reapplication when new policing occurs. Will have to totally rework I'm sure but I think I dwindled social norms/professional talk/intimidating/angry talk that gets challenged/objected to down to a sort of  theoretical core.

**Comment by embray** [comment]
I don't think it should require explaining that calling people "a-holes" and "fascists" is against the spirit laid out by the CoC.

**Comment by ctb** [comment]
On Mon, May 09, 2016 at 05:57:17AM -0700, Erik Bray wrote:
> I don't think it should require explaining that calling people "a-holes" and "fascists" is against the spirit laid out by the CoC.

+inf

**Comment by gvwilson** [comment]
+100

**Comment by SpacedGirl** [comment]
Umm..... @gvwilson? A question? What's the meaning of +inf +100? I'm totally lost on your idea here.

**Comment by ctb** [comment]
It means we strongly endorse the sentiment -- see:

https://www.quora.com/What-does-+1-mean-when-someone-uses-it-in-an-online-forum-or-a-question-website-like-this

**Comment by SpacedGirl** [comment]
@embray and probably almost everyone else.

I'd have to check specifics but not sure but don't think I ever swore at a particular person; just things and concepts.

Also 'fascist' has an appropriate application in so many things; especially at the university given the way things occur these days.

**Comment by SpacedGirl** [comment]
@ctb 
Thanks for the clarification/link. I do get +1 and even +inf but '+inf + 100', definitely not.

**Comment by ctb** [comment]
On Mon, May 09, 2016 at 07:17:12AM -0700, SpacedGirl wrote:
> @ctb 
> Thanks for the clarification/link. I do get +1 and even +inf but '+inf + 100', definitely not.

...those are two different comments, +inf from me for "plus infinity" and
+100 from GVW.

**Comment by SpacedGirl** [comment]
ok. Perhaps it wasn't meant as an additive. The time difference is a little bit close.

**Comment by wrightaprilm** [comment]
Is there more to discuss with the code of conduct, here? If there is, I submit that this thread has gotten too bogged down in one person arguing about the justice of their banning to do that productively, and that the thread should be moved.

**Comment by ctb** [comment]
All: I'd like to focus on specific proposals at this point, if any. A few ideas have been proposed -- lightly skimming the comments,

* @codersquid suggested implementing a few specific strategies [here](https://github.com/swcarpentry/board/issues/111#issuecomment-217483972), that boil down to flipping the moderator bit, touching base with the community, and handling CoC reports with a committee; I'm particularly fond of having a few procedures and scripts sketched out to guide the discussion.
* @wrightaprilm suggested some examples of good/bad behavior, and also brought up the issue of whether a ban on the list applies more broadly to SC/DC spaces [here] (https://github.com/swcarpentry/board/issues/111#issuecomment-217484681) and [here](https://github.com/swcarpentry/board/issues/111#issuecomment-217636270)
* I proposed letting the SCF director (@jduckles) be the initial arbiter, with an appeals board - [here](https://github.com/swcarpentry/board/issues/111#issuecomment-217603918).  I think probably having a few deputies in terms of list moderation and immediate response is not a bad idea - Jonah is not *always* online ;).

Apologies if I didn't capture that all correctly -- please correct me! I personally am happy to entertain any other specific proposals, but I think it's been made clear by @jiffyclub and others that the CoC is here to stay, and that we are exploring ways to enhance its utility and clarity, so would be -1 on any further discussion of that.

I would add that I can't remember more than 2-3 CoC violations in the entire time of the list, since its inception, so I don't know that there is a lot of urgency here in terms of changing anything.

(Just to echo @katyhuff, I have no formal day-to-day role in SCF; just an interested and supportive long-time community member.)

**Comment by jrsmith3** [comment]
> * @wrightaprilm suggested some examples of good/bad behavior, and also brought up the issue of whether a ban on the list applies more broadly to SC/DC spaces here and here 

It seems inconsistent to me to ban someone in one place and not ban them everywhere. If they have violated the CoC, they've violated the CoC.

**Comment by embray** [comment]
:+1: To what Titus wrote.  In particular his first two bullet points are individual actions worth discussing (while the third is a propsoal pertaining to the first bullet point).  Perhaps new issues should be opened for discussing the merits of those individual proposals (1. Moderation guidelines, 2. Examples of good/bad behavior--I'm sure there are already good resources on this that can be pointed to), and this issue should be locked.

Agreed that priority is not that high.

**Comment by swaldman3** [comment]
@ctb One that you've missed: Back in https://github.com/swcarpentry/board/issues/111#issuecomment-217620984, I made the suggestion of a "Welcome to SWC" document that would introduce the community, mention the CoC and the general standard of behavior expected, and also explain things like the roles of the mailing list and github, and generally how to participate. It would effectively form part of the onboarding process (maybe the only part right now) for new volunteers.

Only one person actually responded to this suggestion, though ;-)

**Comment by ctb** [comment]
On Mon, May 09, 2016 at 08:36:58AM -0700, Simon Waldman wrote:
> @ctb One that you've missed: Back in https://github.com/swcarpentry/board/issues/111#issuecomment-217620984, I made the suggestion of a "Welcome to SWC" document that would introduce the community, mention the CoC and the general standard of behavior expected, and also explain things like the roles of the mailing list and github, and generally how to participate. It would effectively form part of the onboarding process (maybe the only part right now) for new volunteers.

+1

> Only one person actually responded to this suggestion, though ;-)

2 now!

**Comment by codersquid** [comment]
@swaldman3 I am a usenetter of yor and liked the periodic faqs. There could be a faq to the mailing list; and/or there could be a welcome email that goes out to new subscribers.

**Comment by swaldman3** [comment]
@codersquid sending it to new subscribers is a good idea (I think the only reason usenet groups did the periodic thing was that there was no way to know about new subscribers). I'd suggest, though, that it should be a page on the website and that what was sent out was a link to it - that way it's also discoverable by prospective volunteers who haven't subscribed. Also, I should note that it should be informal and welcoming - as one of the reasons for suggesting it was to address the concerns of those who found the language of the CoC intimidating.

**Comment by karinlag** [comment]
On 09.05.2016 17:49, Simon Waldman wrote:
> @codersquid <https://github.com/codersquid> sending it to new
> subscribers is a good idea (I think the only reason usenet groups did
> the periodic thing was that there was no way to know about new
> subscribers). I'd suggest, though, that it should be a page on the
> website and that what was sent out was a link to it - that way it's also
> discoverable by prospective volunteers who haven't subscribed. Also, I
> should note that it should be informal and welcoming - as one of the
> reasons for suggesting it was to address the concerns of those who found
> the language of the CoC intimidating.

I am very for having an "Intro to SCF" document that also gets pushed 
with all new email subscriptions. T

Karin

**Comment by swaldman3** [comment]
OK, are we starting to come to some sort of consensus, between my suggestion and those listed by @ctb , that could be reported back to the mailing list?

Re the "Welcome to SWC" document, I'm very happy to draft it, but I would need people who are more closely involved than me to put together a list of things that should be mentioned.

**Comment by tracykteal** [comment]
Thanks @swaldman3 There's definitely consensus that a welcome message should be included when people subscribe to the SWC email lists. 

I put in a PR of a start for such a message
https://github.com/swcarpentry/board/pull/113

Comments welcome.

**Comment by ctb** [comment]
Here's a specific proposal wrt the discuss@ list that (once wording has been reviewed) I think should be passed on to the discuss list and SCF board --

1. In response to a CoC report or violation of other list-specific rules, the current SCF ExecDir (@jduckles) and any appointed deputy moderators have discretion to set the list emergency moderation bit and initiate a ban or user-specific moderation. These actions will be reported back to the list within 24 hours along with the reasoning behind them.
2. Reported CoC violations on the list can be appealed by e-mailing the SCF steering committee at <blah>.
3. A list-specific ban can be extended to any online space managed by the SCF, at the discretion of the Executive Director and the steering committee.

**Comment by ctb** [comment]
All right, it's boring enough that no one has anything to say ;). Created an issue at #114.

**Comment by pipitone** [comment]
With respect to the mailing list, what about including a link on the footer of each message that links to the code of conduct?

**Comment by NelleV** [comment]
Is the CoC a legal document? I see it as guidelines, and if it is anything more, it is illegal in many countries in europe. For instance, in France, you cannot throw someone out of an event  because they haven't followed the SWC CoC, and if they break it badly enough (harassement for example), you are legally bounded to report it.

**Comment by katyhuff** [comment]
It's not a legal document. It's just a policy. The bylaws of the Software
Carpentry Foundation cite it, but even those bylaws are simply an internal
policy of the nonprofit.

**Comment by katyhuff** [comment]
Related : I was surprised codes of conduct could be illegal anywhere. I
know Europython uses one (https://ep2016.europython.eu/en/coc/)

So, I just Googled around and Nelle is (of course) right that it may be
unenforceable in France in some ways ... At least in the case of firing
someone:
https://www.globalregulatoryenforcementlawblog.com/2013/04/articles/government-investigations/france-code-of-conduct-compliance-breach-is-not-automatically-a-sufficient-reason-for-employee-termination-employers-should-be-cautious-of-a-right-local-implementation-of-compliance-guidelines/

Interesting! I would love to see more information on this. It's certainly
enforceable in the US FWIW.

Katy

**Comment by ErinBecker** [comment]
Hi all - please see our new [blog post](http://www.datacarpentry.org/blog/code-of-conduct/) related to this issue.

**Comment by ErinBecker** [comment]
Hi all - in response to community feedback, we've extended the application deadline for the Policy subcommittee to Monday, August 29th at 5pm Pacific (Monday midnight UTC). Looking forward to discussing the many thoughtful comments we've received during our upcoming Lab Meeting - Tuesday, August 16th at 1pm UTC and 10pm UTC.

**Comment by wking** [comment]
Between the policy subcommittee (#146), the updated CoC and workflows (swcarpentry/website#575), the proposed welcome message (#113, still in flight), and the actionable recommendations from this discussion (#114, influenced the CoC updates), I think we can close this issue.  If there's anything else going on in this thread that is *not* addressed by one of those other issues/PRs, it's probably best to open a new issue focused on the remaining bits.

**Comment by jduckles** [comment]
Closing per @wking's suggestion.

**Comment by matthew-brett** [comment]
@SpacedGirl - @davclark just [pointed to](https://github.com/bokeh/bokeh/issues/4854#issuecomment-287810386) your comment [above](https://github.com/swcarpentry/board/issues/111#issuecomment-217748744).  I'd just like to say thank you, and that I think that you expressed the point very well and clearly.

**Comment by davclark** [comment]
Just corrected that comment actual use of @SpacedGirl handle. My apologies for any real and/or perceived oversight.
